src/render.rs

```diff
@@ -345,6 +363,63 @@ impl<'a> Renderer<'a> {
         Ok(output.trim_right().to_string())
     }
 
+    fn render_macro(&mut self, call_node: Node) -> TeraResult<String> {
+        if let MacroCall {namespace, name: macro_name, params: call_params} = call_node {
+            // To find the real namespace we just look at the last one we got.
+            // If we have self, we take the last one otherwise we push it into the vec of namespaces
+            let real_namespace = match namespace.as_ref() {
+                "self" => {
+                    // TODO: handle error if we don't have a namespace
+                    self.macro_namespaces
+                        .last()
```

Comment:
The logic here is somewhat unsettling. It's not clear to me how `macro_namespaces` gets populated the firt time. IE: How does this _not_ panic when calling a recursive macro? In that case the namespace will be `self`, but won't `macro_namespaces` be blank? If not, how does it get its elements?